src/allocation_engine/README.md

```diff
@@ -0,0 +1,135 @@
+# Interval Tree allocation engine
+
+## Design
+
+This interval tree implementation is defined to work as an allocation
+engine for memory-mapped I/O and port I/O address space. It provides
+underlaying methods for allocating, releaseing and updating a memory
+slot. This implementation is going to be used by address allocator
+component of vm-allocator, its main goal is to make address slots
+allocation easier to track and maintain.
+
+```rust
+pub struct IntervalTree {
+    pub(crate) root: Option<InnerNode>,
+}
+
+/// Internal tree node to implement interval tree.
+#[derive(Debug, PartialEq)]
+pub(crate) struct InnerNode {
+    /// Interval handled by this node.
+    pub(crate) key: Range,
+    /// Optional contained data, None if the node is free.
```

Comment:
Probably a leftover from the original implementation when it was `Option`. Now it's `NodeState` enum, so no "Optional" and "None".